packages/template-retail-react-app/app/commerce-api/einstein.js

```diff
@@ -53,6 +53,59 @@ class EinsteinAPI {
         return body
     }
 
+    /**
+     * Given a product or item source, returns the product data that Einstein requires
+     */
+    _constructEinsteinProduct(product) {
+        if (product.type && (product.type.master || product.type.variant)) {
+            // handle variants for PDP / viewProduct
+            // Assumes product is a Product object from SCAPI Shopper-Products:
+            // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-products?meta=type%3AProduct
+            return {
+                id: product.master.masterId,
+                sku: product.id,
+                altId: '',
+                altIdType: ''
+            }
+        } else if (
+            product.productType &&
+            (product.productType.master || product.productType.variant)
+        ) {
+            // handle variants for PLP / viewCategory & viewSearch
+            // Assumes product is a ProductSearchHit from SCAPI Shopper-Search:
+            // https://developer.salesforce.com/docs/commerce/commerce-api/references/shopper-search?meta=type%3AProductSearchHit
+            return {
+                id: product.productId,
+                sku: product.productId, //TODO: Should this be product.representedProduct.id instead?
```

Comment:
I'm interested in some opinions on this. At the moment, for search & category results, I am setting both id and sku to be the product id since at the moment all search results only contain master products.

If we lift that restraint (once we allow support for other product types), I think it may be possible for id and sku to be different? I'm not quite sure what happens once we start returning both master and individual variants in search results.

Reply:
Are we setting the sku as variant id? 🤔

Reply:
Sku id and variant id are the same.

In the case of a variant product, `product.productId` will correspond to the sku id / variant id.
In the case of a master product, `product.productId` will correspond to the master id.